src/comparator.ml

```diff
@@ -0,0 +1,8 @@
+
+type 'a t = 'a -> 'a -> TotalOrder.t
+
+let make (compare : 'a -> 'a -> TotalOrder.t) (this : 'a) (that : 'a): TotalOrder.t =
```

Comment:
I think it should be 
``
let make (compare: 'a -> 'a -> TotalOrder.t) : 'a t = compare
``

I will update the PR later.

Also re. @glennsl .